doc/modules/linear_model.rst

```diff
@@ -810,6 +810,124 @@ loss.
 
     .. [7] Aaron Defazio, Francis Bach, Simon Lacoste-Julien: `SAGA: A Fast Incremental Gradient Method With Support for Non-Strongly Convex Composite Objectives. <https://arxiv.org/abs/1407.0202>`_
 
+.. _Generalized_linear_regression:
+
+Generalized linear regression
+=============================
+
+:class:`GeneralizedLinearRegressor` generalizes the :ref:`elastic_net` in two
+ways [8]_. First, the predicted values :math:`\hat{y}` are linked to a linear
+combination of the input variables :math:`X` via an inverse link function
+:math:`h` as
+
+.. math::    \hat{y}(w, x) = h(xw) = h(w_0 + w_1 x_1 + ... + w_p x_p).
+
+Secondly, the squared loss function is replaced by the deviance :math:`D` of an
+exponential dispersion model (EDM) [9]_. The objective function beeing minimized
+becomes
+
+.. math::    \frac{1}{2s}D(y, \hat{y}) + \alpha \rho ||P_1w||_1
+            +\frac{\alpha(1-\rho)}{2} w^T P_2 w
+
+with sample weights :math:`s`.
+:math:`P_1` (diagonal matrix) can be used to exclude some of the coefficients in
+the L1 penalty, the matrix :math:`P_2` (must be positive semi-definite) allows
+for a more versatile L2 penalty.
+
+Use cases, where a loss different from the squared loss might be appropriate,
+are the following:
+
+  * If the target values :math:`y` are counts (integer valued) or frequencies, you might try a Poisson deviance.
+
+  * If the target values are positive valued and skewed, you might try a Gamma deviance.
+
+  * If the target values seem to be heavier tailed than a Gamma distribution, you might try an Inverse Gaussian deviance (or even higher variance powers of the Tweedie family).
+    Keep in mind that the mean is not a good measure for very heavy tailed distributions, cf. extreme value theory.
```

Comment:
Hm, difficult to find a short answer. Maybe because the world most often is not normally distributed. Two reasons come to my mind:

- Poisson is most natural for counts and frequencies, and Gamma (including the Exponential) is most natural for positive-valued continuous data/processes. The Normal might be most natural for real-valued (positive and negative) continuous data/processes.

- More technically, these loss functions give different weights/importance (in the fitting algo) for large deviations (y_pred vs y_obs) compared to squared loss. In other words, they have different mean-variance relations and are heavier tailed distributions, again compared to the Normal/squared loss. (Heavier tail means large deviations occur more often.)

Should I include any of this reasoning?